framework/aster-frame/src/arch/x86/task/mod.rs

```diff
@@ -0,0 +1,46 @@
+// SPDX-License-Identifier: MPL-2.0
+
+use crate::task::TaskContextApi;
+
+core::arch::global_asm!(include_str!("switch.S"));
+
+#[derive(Debug, Default, Clone, Copy)]
+#[repr(C)]
+pub struct CalleeRegs {
```

Comment:
`TaskContext` is the top-level abstraction provided by this module. `CalleeRegs` is part of it. So put `TaskContext` before `CalleeRegs` so that a reader of this file can see the big picture first.